### ISSUE TITLE
Sign plugins.json

### ISSUE THREAD

**Issue by nashe** [ether/etherpad-lite#3373]
It would be nice to sign https://static.etherpad.org/plugins.json. This way, even in case of compromise of static.etherpad.org, an attacker would not be able to do anything (since we fixed the XSS in https://github.com/ether/etherpad-lite/commit/92eee85f3619a1c0593391bb5dfc57a039b9a4ad). There will be a lot of documentation to write around this feature (who has access to the private key, how to handler revocation, etc), but it's worth it. And it's a first step to an eventual auto-update module ;-)

**Comment by JohnMcLear** [comment]
@gared any thoughts??

**Comment by Gared** [comment]
You mean something like that? https://jose.readthedocs.io/en/latest/
I don't get the idea behind the signing. For now we just offer a list of packages to the users, but honestly we don't know what these plugins are doing (maybe they are doing bad stuff). If someone wants to offer malicious plugins he can do it right now and signing of the plugins.json won't help at all.

Please let me know what do you think about these issues and how these signing process could look like

**Comment by JohnMcLear** [comment]
I agree with @Gared .  

We aren't the Internet police here but perhaps the logic here is a checksum for a plugins filesystem Vs what the update thinks it should be?

This fee

**Comment by JohnMcLear** [comment]
Mm

**Comment by JohnMcLear** [comment]
Final thoughts/note from me.  

Who controls static.etherpad.org other than me?  Namecheap?  Gared?  Mux?

If gared is hit by a bus is all the logic on his box open source/able to be brought back up quickly?

**Comment by Gared** [comment]
@JohnMcLear You control the DNS for static.etherpad.org and I have control over the server. 
The source code for the plugin-updates is in my private repo, but I think I can open it to the world if you want :-)

**Comment by JohnMcLear** [comment]
@gared all active foundation members also have access to dns.  Currently your server would be where things fell apart of something went wrong.

Both the source code for the service and access to the hosting needs to be available to the foundation members.  The foundation will obviously pick up the bill for the server etc.  I wonder if heroku or something would suffice or maybe we could use git to serve the plugins JSON file and a different web front end for static etherpad.org web UI?

**Comment by Gared** [comment]
@JohnMcLear No etherpad itself uses the plugins.json. There is no simple way in npm to list all packages with a prefix.
I will publish the source code for the backend and then we can decide where we want to host it

**Comment by JohnMcLear** [comment]
Okay perfect.  Thanks

**Comment by Gared** [comment]
Published the code here: https://github.com/ether/plugin-loader

**Comment by JohnMcLear** [comment]
Awesome ta!  Thinking of building in some of the checkPlugins.js I built.  Any thoughts on that?

**Comment by Gared** [comment]
The checkPlugins.js is mainly for plugin developers and the goal of the plugin-page is to give etherpad-admins an overview about the plugins. In my opinion this script is not directly something for the plugin-page, but I started a few weeks ago to parse the ep.json and see what hooks the plugin is using and displaying icons at each plugin. Sadly most of the hooks changes something in the frontend and it's hard to group these hooks to display anything useful to the user on the plugin-page. Maybe you have any ideas? Maybe we should discuss this topic in an issue in the right repository

**Comment by JohnMcLear** [comment]
@Gared agreed we should take this topic into the right repo.  Will frame it now.

**Comment by JohnMcLear** [comment]
Closing due to inactivity — this issue has had minimal engagement and no recent progress. If this is still relevant, please reopen with updated context.
